hack/verify-testing-import.sh

```diff
@@ -40,21 +41,23 @@ RELEASE_BIN_PKGS=(
   "${KUBE_ROOT}/cmd/kubectl-convert"
   "${KUBE_ROOT}/cmd/kubelet"
   "${KUBE_ROOT}/cmd/kubeadm"
+  # code generators
+  "${KUBE_ROOT}"/staging/src/k8s.io/code-generator/cmd/*/
 )
 
 pkgs_with_testing_import=()
-for file in "${RELEASE_BIN_PKGS[@]}"
+for pkg in "${BIN_PKGS[@]}"
 do
-  if [ "$(go list -json "${file}" | jq 'any(.Deps[]; . == "testing")')" == "true" ]
+  if [ "$(go list -json "${pkg}" | jq 'any(.Deps[]; . == "testing")')" == "true" ]
```

Comment:
in addition to not checking the code-generator packages, I think the main issue with https://github.com/kubernetes/kubernetes/issues/131646 was that we didn't catch our `k8s.io/client-go/testing` package as a "testing" package

Reply:
I think that's only because we only checked this list of release binaries. Since the package name is "testing" it should've been caught here (like kubemark would be for cadvisor testing)

I'll verify that though.

Reply:
I see now. Patching the jq

Reply:
looks like these two:

* staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/generic/policy_test_context.go, only used by tests, should move to a testing subpackage or something
* cmd/kubeadm/app/util/apiclient/dryrun.go, this will be harder to unwind, looks like it is widely used

maybe put the client-go/testing check in a separate loop with its own list of binaries to check, and exclude kubeadm from that check for now, and file an issue for cluster-lifecycle to resolve

Reply:
kubelet also had an easy to resolve testing import https://github.com/kubernetes/kubernetes/pull/131750/commits/8d9f0f0db32fb12c05f74a8108b6f2963b785d00

Reply:
> looks like these two:

I'm seeing even after dropping the revert commits:
```
Testing packages are imported in:
  ./cmd/cloud-controller-manager: k8s.io/client-go/testing
  ./cmd/kube-apiserver: k8s.io/client-go/testing
  ./cmd/kube-controller-manager: k8s.io/client-go/testing
  ./cmd/kube-proxy: k8s.io/client-go/testing
  ./cmd/kube-scheduler: k8s.io/client-go/testing
  ./cmd/kubectl: k8s.io/client-go/testing
  ./cmd/kubectl-convert: k8s.io/client-go/testing
  ./cmd/kubelet: k8s.io/client-go/testing
  ./cmd/kubeadm: k8s.io/client-go/testing
```

Reply:
> staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/generic/policy_test_context.go, only used by tests, easy to move to a testing subpackage or something

this is not so easy to move, there's a circular dependency problem for policy_source_test.go

Reply:
hrmm... policy_source_test.go uses an isolated testing package which avoids the circular dependency

it looks like https://github.com/liggitt/kubernetes/commits/testingisolate/ gets close, but the last hangup might be use of the private guts of `type policySource` from inside `policy_test_context.go`

agree that will not be trivial to unwind, but is probably worth doing at background priority

Reply:
Does your branch build?
You have the generictesting package using types from generic but the generic package using generictesting to test.

I tried moving the test cases but they're testing internals.

We could split into a third internal package for policy source + tests and re-export the public parts.

Reply:
> Does your branch build?

no, I thought I got it down to just a couple compile issues but maybe that was masking a whole other set of issues